lib/SILGen/SILGen.cpp

```diff
@@ -803,10 +803,13 @@ emitMarkFunctionEscapeForTopLevelCodeGlobals(SILLocation loc,
     TopLevelSGF->B.createMarkFunctionEscape(loc, Captures);
 }
 
-void SILGenModule::emitAbstractFuncDecl(AbstractFunctionDecl *AFD) {
+bool SILGenModule::emitAbstractFuncDecl(AbstractFunctionDecl *AFD) {
   // Emit any default argument generators.
   emitDefaultArgGenerators(AFD, AFD->getParameters());
 
+  if (AFD->wasBodyTypecheckingSkipped())
```

Comment:
Instead, I would prefer it if the skipping was done further up the callstack. Returning a boolean is subtle

Reply:
... or further down. See how emitOrDelayFunction takes a SILDeclRef and a callback? If the SILDeclRef's isSerialized() is not set, you can just not call the callback and return. This will handle default args, accessors, functions, and most other things, except for vtable and witness thunks.